### PR TITLE
Docs: Plugin doc review for chunk 1-B

### PR DESCRIPTION
**Pull request by josmperez**
**What is this feature?**

This is the first of two PRs holding a review of plugins docs, focusing on structure, editorial soundness, and conformity to doc team style guidelines.

Also, the page's voice and tone in a few places are enthusiastic and upbeat. Suggestions on voice and tone and overall contents are welcome.

**Why do we need this feature?**

This fulfills the doc improvement initiative described in the plugins doc plan.

**Who is this feature for?**

Plugin developers

### CODE REVIEW

**Review by andresmgot — COMMENTED** [review]
I am not sure what relation is there between this PR and https://github.com/grafana/grafana/pull/67070. Maybe you wanted to split the changes in two PRs? If that's the case, you need to send the PR from a clean branch (`main`) and include only commits that were not already in the other branch.

File: docs/sources/developers/plugins/_index.md [hunk 1]
```diff
@@ -6,77 +6,51 @@ title: Build a plugin
...
-- [Build a data source backend plugin](/tutorials/build-a-data-source-backend-plugin/)
-
-Ready to learn more? Check out our other tutorials:
+## Contents of this developer's guide
 
-- [Build a panel plugin with D3.js](/tutorials/build-a-panel-plugin-with-d3/)
+The following topics give you an idea of what to expect from this guide:
 
-### Guides
+- [**Get started with plugins** - The easiest way to start developing Grafana plugins is to use the Create-plugin tool. Walk step by step for downloading the tool and using it to get your plugin up and running as fast as possible.
```
Comment (academo) [inline docs/sources/developers/plugins/_index.md:34]: please add a link to the tool.

File: docs/sources/developers/plugins/_index.md [hunk 2]
```diff
@@ -6,77 +6,51 @@ title: Build a plugin
...
-Open the terminal, and run the following command in your [plugin directory]({{< relref "../../setup-grafana/configure-grafana/#plugins" >}}):
+## Plugin basics
 
-```bash
-npx @grafana/create-plugin@latest
-```
+These are among the types of plugins you can create:
 
-Follow the questions and you will have a starter plugin ready to develop.
+- **Panel plugins** - Visualize data and navigating between dashboards.
```
Comment (academo) [inline docs/sources/developers/plugins/_index.md:21]: can we add links to the tutorials to create these kind of plugins?

File: docs/sources/developers/plugins/_index.md [hunk 3]
```diff
@@ -6,77 +6,51 @@ title: Build a plugin
...
-
-Ready to learn more? Check out our other tutorials:
+## Contents of this developer's guide
 
-- [Build a panel plugin with D3.js](/tutorials/build-a-panel-plugin-with-d3/)
+The following topics give you an idea of what to expect from this guide:
 
-### Guides
+- [**Get started with plugins** - The easiest way to start developing Grafana plugins is to use the Create-plugin tool. Walk step by step for downloading the tool and using it to get your plugin up and running as fast as possible. 
+- [**Introduction to plugin development** - Ground yourself by learning the key concepts involved in Grafana plugin development. For example, learn about frontend and backend development processe, data frames, and error handling.
```
Comment (academo) [inline docs/sources/developers/plugins/_index.md:35]: Shouldn't all these elements be links to their respective page?

File: docs/sources/developers/plugins/publish-a-plugin.md [hunk 4]
```diff
@@ -7,140 +7,39 @@ title: Publish a plugin
...
+## Before you begin
 
-**Can I see metrics of my plugin installs, downloads or usage?**
+Before you begin to publish a plugin, it needs to be built and packaged. 
 
-- No. We don't offer this information at the moment to plugin authors.
+To build a plugin from source, Grafana creates a `dist` directory that contains the production build, or _plugin assets_, for your plugin.
 
-**How can I update my plugin's catalog page?**
+When the Grafana server starts, it recursively looks in the plugin directory for any directory that contains a `plugin.json` file, and then it tries to load the plugin assets in the same directory.
```
Comment (academo) [inline docs/sources/developers/plugins/publish-a-plugin.md:24]: I don't see this sentence fitting in this context.

**Comment by jdbaldry** [comment]
@josmperez I just pushed a commit that renamed `publish-a-plugin` to `publish-a-plugin.md` because it would break the `make docs` builds.

File: docs/sources/developers/plugins/publish-a-plugin.md [hunk 5]
```diff
@@ -165,9 +64,9 @@ For every new plugin, we perform a manual review that includes the following che
   If you're unable to make the source code publicly available, let us know in a comment on your plugin submission.
 - **Tests:** We install your plugin on one of our Grafana instances to test it for basic use.
   For more advanced plugins, we may ask you to assist us in configuring a test environment for the plugin.
-  This test environment will be used whenever you submit an plugin update.
+  We will use the test environment whenever you submit a plugin update.
```
Comment (imatwawana) [inline docs/sources/developers/plugins/publish-a-plugin.md:67]:
```suggestion
  We'll use the test environment whenever you submit a plugin update.
```

File: docs/sources/developers/plugins/publish-a-plugin.md [hunk 6]
```diff
@@ -180,6 +79,47 @@ To submit an **update** for an already published plugin:
...
+
+**How long does it take to review my submission?**
+
+- We're not able to give an estimate at this time, though we're constantly working on improving the time it takes to review a plugin.
+
+**Can I decide a date when my plugin will be published?**
+
+- No. We cannot guarantee specific publishing dates, as plugins are immediately published after a review based on our internal prioritization.
+
+**Can I see metrics of my plugin installs, downloads or usage?**
```
Comment (imatwawana) [inline docs/sources/developers/plugins/publish-a-plugin.md:100]:
```suggestion
**Can I see metrics of my plugin installs, downloads, or usage?**
```

File: docs/sources/developers/plugins/publish-a-plugin.md [hunk 7]
```diff
@@ -180,6 +79,47 @@ To submit an **update** for an already published plugin:
...
+
+- No. We don't offer this information at the moment to plugin authors.
```
Comment (imatwawana) [inline docs/sources/developers/plugins/publish-a-plugin.md:102]:
```suggestion
- No. We don't offer this information to plugin authors at the moment.
```

File: docs/sources/developers/plugins/publish-a-plugin.md [hunk 8]
```diff
@@ -180,6 +79,47 @@ To submit an **update** for an already published plugin:
...
+
+- In the event of a bug, unlisting the plugin from our catalog may be possible in exceptional cases, such as security concerns. However, we don't have control over the specific instances where the plugin is installed.
+
+**Can I distribute my plugin somewhere else other than the Grafana plugin catalog?**
+
+- The official method for distributing Grafana plugins is through our catalog. Alternative methods, such as installing private or development plugins on local Grafana instances, are available as per the guidelines provided in [this guide]({{< relref "../../administration/plugin-management#install-plugin-on-local-grafana" >}}).
+
+**Can I still use Angular for my plugin?**
+
+- No. We will not accept any new plugin submissions written in angular. For more information, refer to our [Angular support deprecation documentation]({{< relref "../angular_deprecation" >}}).
```
Comment (imatwawana) [inline docs/sources/developers/plugins/publish-a-plugin.md:119]:
```suggestion
- No. We won't accept any new plugin submissions written in Angular. For more information, refer to our [Angular support deprecation documentation]({{< relref "../angular_deprecation" >}}).
```

File: docs/sources/developers/plugins/publishing-and-signing-criteria.md [hunk 9]
```diff
@@ -6,26 +6,37 @@ title: Plugin publishing and signing criteria
...
-  - Should not access the filesystem
-  - Should not access environment variables
+### Data collection
+Plugins cannot collect usage or user information. Violations of this guideline include but are not limited to:
+   - Directly collecting installation and user statistics.
+   - Sending data to third parties for analytics purposes.
+   - Embedding tracking code.
+
+### Data at rest
+Sensitive data such as credentials and user information must be encrypted using industry standards.
```
Comment (imatwawana) [inline docs/sources/developers/plugins/publishing-and-signing-criteria.md:20]:
```suggestion
Sensitive data such as credentials and user information must be encrypted using industry standards:
```

File: docs/sources/developers/plugins/publishing-and-signing-criteria.md [hunk 10]
```diff
@@ -6,26 +6,37 @@ title: Plugin publishing and signing criteria
...
+   - Use secureJsonData to store data source credentials.
+   - Secrets cannot be stored in panel options.
+
+### Data transmission
+Secure methods that meet industry standard encryption levels should be used, such as Secure Sockets Layer (SSL) or Transport Layer Security (TLS).
```
Comment (imatwawana) [inline docs/sources/developers/plugins/publishing-and-signing-criteria.md:25]:
```suggestion
Secure methods that meet industry standard encryption levels should be used, such as Secure Sockets Layer (SSL) or Transport Layer Security (TLS):
```

File: docs/sources/developers/plugins/publishing-and-signing-criteria.md [hunk 11]
```diff
@@ -6,26 +6,37 @@ title: Plugin publishing and signing criteria
...
+
+### Abuse
+Plugins should not perform actions beyond the scope of the intended use.
```
Comment (imatwawana) [inline docs/sources/developers/plugins/publishing-and-signing-criteria.md:28]:
```suggestion
Plugins should not perform actions beyond the scope of the intended use:
```

File: docs/sources/developers/plugins/publishing-and-signing-criteria.md [hunk 12]
```diff
@@ -6,26 +6,37 @@ title: Plugin publishing and signing criteria
...
+   - Do not include hidden files.
+   - Do not manipulate the underlying environment, privileges, or related processes.
+
+### Security
+Plugins should not access system areas beyond the scope of the intended use.
```
Comment (imatwawana) [inline docs/sources/developers/plugins/publishing-and-signing-criteria.md:33]:
```suggestion
Plugins should not access system areas beyond the scope of the intended use:
```

File: docs/sources/developers/plugins/publishing-and-signing-criteria.md [hunk 13]
```diff
@@ -36,8 +47,6 @@ Plugins must be licensed under one of the following AGPL compliant licenses for
 
 If contributing a plugin on behalf of an organization, be sure to seek guidance from your legal team.
 
-## Commercial
-
-- Usage of 3rd party software or dependencies within the plugin must be licensed for the intended use. For example, using open source dependencies must be credited/licensed; embedding logos or trademarks;
+## Commercial licensing
 
-Grafana Labs reserves the right to decline or remove any plugin at its discretion. Failure to comply with publishing and signing criteria may result in immediate removal from the Grafana plugin catalog.
+Usage of third party software or dependencies within the plugin must be licensed for the intended use. For example, using open source dependencies must be credited/licensed; and embedding logos or trademarks.
```
Comment (imatwawana) [inline docs/sources/developers/plugins/publishing-and-signing-criteria.md:52]: Not sure what the bit after the semi-colon means

**Review by imatwawana — COMMENTED** [review]
I did a pass on this, but when I started this review, I wasn't able to build the branch locally because of the file that was missing the .md, thus, I haven't tested any links. I can take a look at this again, but probably not for a couple days.

**Comment by josmperez** [comment]
As reviewers pointed out, this PR contained files that it shouldn't have. I have created a new PR built from main that fixes this issue. Closing.

File: docs/sources/developers/plugins/package-a-plugin [hunk 14]
```diff
@@ -0,0 +1,64 @@
+---
```
[the remark this answers is not in the document]
Comment (josmperez) [reply docs/sources/developers/plugins/package-a-plugin]: This PR is now closed. This issue is fixed in PR 67164.

File: docs/sources/developers/plugins/publish-a-plugin.md [hunk 15]
```diff
@@ -7,140 +7,39 @@ title: Publish a plugin
...
+By having your plugin published on Grafana.com, more users will be able to discover your plugin.
 
-- No. We cannot guarantee specific publishing dates, as plugins are immediately published after a review based on our internal prioritization.
+## Before you begin
 
-**Can I see metrics of my plugin installs, downloads or usage?**
+Before you begin to publish a plugin, it needs to be built and packaged. 
 
-- No. We don't offer this information at the moment to plugin authors.
+To build a plugin from source, Grafana creates a `dist` directory that contains the production build, or _plugin assets_, for your plugin.
```
[the remark this answers is not in the document]
Comment (josmperez) [reply docs/sources/developers/plugins/publish-a-plugin.md:22]: This PR is now closed. This issue is fixed in PR 67164.

File: docs/sources/developers/plugins/publish-a-plugin.md (hunk 4 above)
Comment (josmperez) [reply docs/sources/developers/plugins/publish-a-plugin.md:24]: This PR is now closed. This issue is fixed in PR 67164.

File: docs/sources/developers/plugins/_index.md (hunk 2 above)
Comment (josmperez) [reply docs/sources/developers/plugins/_index.md:21]: Let's hold off. This section is probably going to be moving to a separate topic on types of plugins soon. As well, we don't have tutorials for all the types in this list at this time.

File: docs/sources/developers/plugins/_index.md [hunk 16]
```diff
@@ -6,77 +6,51 @@ title: Build a plugin
...
+These are among the types of plugins you can create:
 
-Follow the questions and you will have a starter plugin ready to develop.
+- **Panel plugins** - Visualize data and navigating between dashboards.
+- **Data source plugins** - Link to new databases or other sources of data.
+- **App plugins** - Bundle panels and data sources together to create a better user experience.
+- **Other plugins** - Visualize data source logs or streaming data sources.
 
-If you want a more guided introduction to plugin development, check out our tutorials:
+The possibilities are as endless as your imagination, and you can use this guide to help make your dreams become reality.
```
[the remark this answers is not in the document]
Comment (josmperez) [reply docs/sources/developers/plugins/_index.md:26]: 100% correct. I have axed this is the current version.

File: docs/sources/developers/plugins/_index.md (hunk 1 above)
Comment (josmperez) [reply docs/sources/developers/plugins/_index.md:34]: Done in PR 67070.

File: docs/sources/developers/plugins/_index.md (hunk 3 above)
Comment (josmperez) [reply docs/sources/developers/plugins/_index.md:35]: Done in PR 67070.

File: docs/sources/developers/plugins/_index.md [hunk 17]
```diff
@@ -6,77 +6,51 @@ title: Build a plugin
...
+- **Data source plugins** - Link to new databases or other sources of data.
+- **App plugins** - Bundle panels and data sources together to create a better user experience.
+- **Other plugins** - Visualize data source logs or streaming data sources.
 
-If you want a more guided introduction to plugin development, check out our tutorials:
+The possibilities are as endless as your imagination, and you can use this guide to help make your dreams become reality. 
 
-- [Build a panel plugin](/tutorials/build-a-panel-plugin/)
-- [Build a data source plugin](/tutorials/build-a-data-source-plugin/)
+   Note: To learn more about the types of plugins you can build, refer to the [Plugin Overview]({{< relref "../../administration/plugin-management/" >}}) documentation.
```
[the remark this answers is not in the document]
Comment (josmperez) [reply docs/sources/developers/plugins/_index.md:28]: Done in PR 67070.

File: docs/sources/developers/plugins/_index.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (josmperez) [reply docs/sources/developers/plugins/_index.md:34]: Done in PR 67070.

File: docs/sources/developers/plugins/_index.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (josmperez) [reply docs/sources/developers/plugins/_index.md:35]: Done in PR 67070.

File: docs/sources/developers/plugins/_index.md [hunk 18]
```diff
@@ -6,77 +6,51 @@ title: Build a plugin
...
-Ready to learn more? Check out our other tutorials:
+## Contents of this developer's guide
 
-- [Build a panel plugin with D3.js](/tutorials/build-a-panel-plugin-with-d3/)
+The following topics give you an idea of what to expect from this guide:
 
-### Guides
+- [**Get started with plugins** - The easiest way to start developing Grafana plugins is to use the Create-plugin tool. Walk step by step for downloading the tool and using it to get your plugin up and running as fast as possible. 
+- [**Introduction to plugin development** - Ground yourself by learning the key concepts involved in Grafana plugin development. For example, learn about frontend and backend development processe, data frames, and error handling. 
+- [**Create a Grafana plugin** - If you are already familiar with plugin creation basics, then it's time to dive deeper into specific development scenarios. You will find tutorials for creating panel plugins, data source plugins, and more.
```
[the remark this answers is not in the document]
Comment (josmperez) [reply docs/sources/developers/plugins/_index.md:36]: Done in PR 67070.

File: docs/sources/developers/plugins/build-a-logs-data-source-plugin.md [hunk 19]
```diff
@@ -51,7 +58,9 @@ That's all you need to start returning log data from your data source. Go ahead
 
 Congratulations, you just wrote your first logs data source plugin! Next, let's look at a couple of features that can further improve the experience for the user.
 
-### (Optional) Add preferred visualisation type hint to the data frame
+## Enhance your logs data source plugin with optional features
+
```
[the remark this answers is not in the document]
Comment (josmperez) [reply docs/sources/developers/plugins/build-a-logs-data-source-plugin.md:62]: Done in PR 67070.

File: docs/sources/developers/plugins/build-a-logs-data-source-plugin.md [hunk 20]
```diff
@@ -51,7 +58,9 @@ That's all you need to start returning log data from your data source. Go ahead
...
+### Add a preferred visualisation type hint to the data frame
```
[the remark this answers is not in the document]
Comment (josmperez) [reply docs/sources/developers/plugins/build-a-logs-data-source-plugin.md:63]: Done in PR 67070.

File: docs/sources/developers/plugins/_index.md [hunk 21]
```diff
@@ -6,77 +6,51 @@ title: Build a plugin
...
+- [**Introduction to plugin development** - Ground yourself by learning the key concepts involved in Grafana plugin development. For example, learn about frontend and backend development processe, data frames, and error handling. 
+- [**Create a Grafana plugin** - If you are already familiar with plugin creation basics, then it's time to dive deeper into specific development scenarios. You will find tutorials for creating panel plugins, data source plugins, and more. 
+- [**Migrate a plugin** - Learn how to upgrade from a previous version of a Grafana plugin, rewrite an old Angular plugin in React, or update to a newer version. 
+- [**Publish a Grafana plugin**  - Here you'll find what you need to know if you want to publish a plugin to the Grafana Catalog, including publishing criteria, signing process, and publishing.
+- [**Advanced topics** - Learn all about advanced plugin features such as nested plugins. If you find yourself needing to work with Legacy plugins, check here for recommendations.
+- [**Reference** - Description of the Plugin.json schema and plugin metadata.
```
[the remark this answers is not in the document]
Comment (josmperez) [reply docs/sources/developers/plugins/_index.md:40]: Done in PR 67070.

File: docs/sources/developers/plugins/build-a-streaming-data-source-plugin.md [hunk 22]
```diff
@@ -6,27 +6,35 @@ title: Build a streaming data source plugin
...
-For example, a streaming data source plugin can connect to a websocket, or subscribe to a message bus, and update the visualization whenever a new message is available.
+## Add _streaming_ to your data source
+
+Enable _streaming_ for your data source plugin to update your dashboard as soon as new data becomes available.
 
-Let's see how you can add streaming to an existing data source!
+For example, a streaming data source plugin can connect to a websocket, or subscribe to a message bus, and update the visualization whenever a new message is available.
 
-Grafana uses [RxJS](https://rxjs.dev/) to continuously send data from a data source to a panel visualization. There's a lot more to RxJS than what's covered in this guide. If you want to learn more, check out the [RxJS documentation](https://rxjs.dev/guide/overview).
+### Step 1: Edit the plugin.json file
```
[the remark this answers is not in the document]
Comment (josmperez) [reply docs/sources/developers/plugins/build-a-streaming-data-source-plugin.md:25]: Done in PR 67070.

File: docs/sources/developers/plugins/data-frames.md [hunk 23]
```diff
@@ -26,45 +24,47 @@ interface DataFrame {
...
 
 Each field in a data frame contains optional information about the values in the field, such as units, scaling, and so on.
 
 By adding field configurations to a data frame, Grafana can configure visualizations automatically. For example, you could configure Grafana to automatically set the unit provided by the data source.
 
-## Transformations
+## Data transformations
 
-Along with the type information, field configs enable _data transformations_ within Grafana.
+We have seen how field configs contain type information, and they also have another role. Data frame fields enable _data transformations_ within Grafana.
```
[the remark this answers is not in the document]
Comment (josmperez) [reply docs/sources/developers/plugins/data-frames.md:67]: Done in PR 67070.

File: docs/sources/developers/plugins/data-frames.md [hunk 24]
```diff
@@ -161,11 +161,11 @@ Dimensions: 5 fields by 2 rows
...
 ```
 
-> **Note:** Not all panels support the wide time series data frame format. To keep full backward compatibility we have introduced a transformation that can be used to convert from the wide to the long format. For usage information, refer to the [Prepare time series-transformation]({{< relref "../../panels-visualizations/query-transform-data/transform-data/#prepare-time-series" >}}).
+> **Note:** Not all panels support the wide time series data frame format. To keep full backward compatibility Grafana has introduced a transformation that you can use to convert from the wide to the long format. For usage information, refer to the [Prepare time series-transformation]({{< relref "../../panels-visualizations/query-transform-data/transform-data/#prepare-time-series" >}}).
 
 ## Technical references
 
-This section contains links to technical reference and implementations of data frames.
+The concept of a data frame in Grafana is borrowed from data analysis tools like the [R programming language](https://www.r-project.org), and [Pandas](https://pandas.pydata.org/). Other technical references are provided below.
```
[the remark this answers is not in the document]
Comment (josmperez) [reply docs/sources/developers/plugins/data-frames.md:168]: Done in PR 67070.

File: docs/sources/developers/plugins/publish-a-plugin.md (hunk 6 above)
Comment (josmperez) [reply docs/sources/developers/plugins/publish-a-plugin.md:100]: Done in PR 67070.

File: docs/sources/developers/plugins/publish-a-plugin.md (hunk 7 above)
Comment (josmperez) [reply docs/sources/developers/plugins/publish-a-plugin.md:102]: Done in PR 67070.

File: docs/sources/developers/plugins/publish-a-plugin.md (hunk 8 above)
Comment (josmperez) [reply docs/sources/developers/plugins/publish-a-plugin.md:119]: Done in PR 67070.

File: docs/sources/developers/plugins/publishing-and-signing-criteria.md (hunk 9 above)
Comment (josmperez) [reply docs/sources/developers/plugins/publishing-and-signing-criteria.md:20]: Done in PR 67070.

File: docs/sources/developers/plugins/publishing-and-signing-criteria.md (hunk 10 above)
Comment (josmperez) [reply docs/sources/developers/plugins/publishing-and-signing-criteria.md:25]: Done in PR 67070.

File: docs/sources/developers/plugins/publishing-and-signing-criteria.md (hunk 11 above)
Comment (josmperez) [reply docs/sources/developers/plugins/publishing-and-signing-criteria.md:28]: Done in PR 67070.

File: docs/sources/developers/plugins/publishing-and-signing-criteria.md (hunk 12 above)
Comment (josmperez) [reply docs/sources/developers/plugins/publishing-and-signing-criteria.md:33]: Done in PR 67070.

File: docs/sources/developers/plugins/publishing-and-signing-criteria.md (hunk 13 above)
Comment (josmperez) [reply docs/sources/developers/plugins/publishing-and-signing-criteria.md:52]: Done in PR 67070.

**Pull request closed without merging**